modin/core/execution/ray/generic/io/io.py

```diff
@@ -196,3 +197,77 @@ def func(df, **kw):
         for rows in result:
             for partition in rows:
                 wait([partition.oid])
+
+    @staticmethod
+    def _to_parquet_check_support(kwargs):
+        """
+        Check if parallel version of ``to_parquet`` could be used.
+
+        Parameters
+        ----------
+        kwargs : dict
+            Keyword arguments passed to ``.to_parquet()``.
+
+        Returns
+        -------
+        bool
+            Whether parallel version of ``to_parquet`` is applicable.
+        """
+        path = kwargs["path"]
+        compression = kwargs["compression"]
+        if not isinstance(path, str):
+            return False
+        if any((path.endswith(ext) for ext in [".gz", ".bz2", ".zip", ".xz"])):
+            return False
+        if compression is None or not compression == "snappy":
+            return False
+        return True
+
+    @classmethod
+    def to_parquet(cls, qc, **kwargs):
+        """
+        Write a ``DataFrame`` to the binary parquet format.
+
+        Parameters
+        ----------
+        qc : BaseQueryCompiler
+            The query compiler of the Modin dataframe that we want to run ``to_parquet`` on.
+        **kwargs : dict
+            Parameters for ``pandas.to_parquet(**kwargs)``.
+        """
+        if not cls._to_parquet_check_support(kwargs):
+            return BaseIO.to_parquet(qc, **kwargs)
+
+        def func(df, **kw):
+            """
+            Dump a chunk of rows as parquet, then save them to target maintaining order.
+
+            Parameters
+            ----------
+            df : pandas.DataFrame
+                A chunk of rows to write to a parquet file.
+            **kw : dict
+                Arguments to pass to ``pandas.to_parquet(**kw)`` plus an extra argument
+                `partition_idx` serving as chunk index to maintain rows order.
+            """
+            output_path = kwargs["path"]
+            compression = kwargs["compression"]
+            partition_idx = kw["partition_idx"]
+            if not os.path.exists(output_path):
+                os.makedirs(output_path)
+            kwargs[
+                "path"
+            ] = f"{output_path}/part-{partition_idx:04d}.{compression}.parquet"
+            df.to_parquet(**kwargs)
+            return pandas.DataFrame()
+
+        # signaling that the partition with id==0 can be written to the file
```

Comment:
I guess you copied the code of recent implementation of `read_csv`. Please, make sure that parallel `to_parquet` is correct.

Reply:
Yes, `to_parquet` is writing to files in parallel and I have tested both manually and the ray tests on TeamCity pass.

Reply:
@YarShev it is well tested at this point.

Reply:
What is the comment above for if we don't perform any synchronization. Actually, why don't we need to perform the synchronization?

Reply:
Ah that comment is not necessary, removed it. Parquet is a serialized way of storing a Pandas DataFrame (i.e. you are writing the exact representation from the dataframe to disk so the types of columns and indices are the same). So a synchronization process for different partitions is not necessary.

Reply:
Okay, I see